src/caching.jl

```diff
@@ -1,12 +1,15 @@
-const CGCKEY{N} = NTuple{3,SUNIrrep{N}}
-const CGCCache{N,T} = LRU{CGCKEY{N},SparseArray{T,4}}
+const CGCKey{N} = NTuple{3,SUNIrrep{N}}
+
+"""
+    CGC_CACHE = LRU{CGCKey,SparseArray{Float64,4}}(; maxsize=100_000)
+
+Global cache for storing Clebsch-Gordan Coefficients.
+"""
+const CGC_CACHE = LRU{CGCKey,SparseArray{Float64,4}}(; maxsize=100_000)
```

Comment:
Is the use of an abstract key type useful here, as opposed to just using `Any`? The definition looks nicer with this specialisation in place, but I remember reading somewhere (long time ago) that container types which are not concrete types could just as well be `Any` (and that this choice might be advantageous because more methods would already be compiled for it in the sysimg).

Reply:
I didn't check to be honest. It used to be a cache for each value of `N`, which was then concretely typed, but I don't think it really matters here. The cost of the cache lookup should be negligible compared to actually using the CGCs anyways, so I mostly don't want to spend too much time optimizing this. I'll change it to `Any` to be consistent with other places in the code, but in the future I would really like to migrate the caching functionality to `TensorKitSectors`, such that sectors can automatically opt in to both RAM and disk caching with various strategies.

Reply:
Having a general caching functionality in TensorKitSectors.jl sounds like a great idea!